enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,636 @@
+
+There needs to be extra validation in place in such a way that
+using the `measured` SCC will also result in workloads
+only being able to use profiles that are installed in the same
+namespace as the workload.
+
+This will prevent workloads trying to:
+
+* Wrongly use profiles that are assumed to be "pre-installed" in the system.
+
+* Share a profile accross namespaces which might lead to errors.
+
+* Maliciously gain privileges by using a profile installed from another
+  namespace.
+
+### OLM integration
+
+As part of OpenShift promoting security as part of the platform, application
+developers need to be able to install Security Profiles when installing operators.
+
+For this, a `securityProfiles` section could be added to the CSV definition as part
+of the `install.spec` section. Following a similar pattern as permissions, it could
+look as follows:
+
+```yaml
+...
+  install:
+    spec:
+      clusterPermissions:
+        - rules:
+          - apiGroups:
+            - ""
+            resources:
+            - nodes
+            verbs:
+            - get
+            - list
+            - watch
+          serviceAccountName: some-sa
+      securityProfiles:
+        - seccomp:
+            spec:
+              defaultAction: "SCMP_ACT_LOG"
+          serviceAccountName: some-sa
+        - selinux:
+            spec:
+              policy: |
+                (blockinherit container)
+                (allow process var_log_t ( dir ( open read getattr lock search ioctl add_name remove_name write ))) 
+                (allow process var_log_t ( file ( getattr read write append ioctl lock map open create  ))) 
+                (allow process var_log_t ( sock_file ( getattr read write append open  ))
+          serviceAccountName: some-sa
+      deployments:
+      - name: some-deployment
+        spec:
+          replicas: 1
+      ...
+
+```
+
+Where SELinux and Seccomp profile definitions are linked to a
+Service Account, similar to other objects in the CSV.
+
+### User Stories
+
+The personas it aims to help are documented in [the project's documentation](
+https://github.com/kubernetes-sigs/security-profiles-operator/blob/master/doc/personas.md).
+
+The user stories that are being covered are [also appropriately documented](
+https://github.com/kubernetes-sigs/security-profiles-operator/blob/master/doc/user-stories.md).
+
+### Implementation Details/Notes/Constraints
+
+#### selinuxd
+
+Selinuxd is currently deployed as a container alongside the SPOD instance.
+In order to ensure that SELinux policies are installed early on when a
+node boots, it would be ideal to have selinuxd run directly on the node
+itself (either as a systemd unit or as a static container) as opposed to
+it running on a container managed by the Kubelet.
+
+On the other hand, the repository namespace selinuxd is hosted is currently
+a personal one. It would also be ideal to move that somewhere more appropriate.
+(The coreos namespace maybe?)
```

Comment:
@cgwalters @cverna I would like to get your thoughts on where should selinuxd be deployed.

I think that having selinuxd as a daemon on the node would be best, because then any policies dropped in via MCO would be installed before workloads start. But it is another package on the OS and in addition a daemon that always runs..

A static pod would also be a possibility, selinuxd is capable of running in a container. What I see as a benefit of a deamon on the node is that we would be able to install policies also for static pods if selinuxd is configured to be started before kubelet.

If neither of these is acceptable, selinuxd could just remain part of the SPO and run as a deamonset. But then policies might be installed later than a workload is started..

Note that selinuxd is usable even w/o SPO as a way to take SELinux policies into use if an admin distributes them via an MCO or some other mechanism.

At the moment, selinuxd lives in a personal GH repo and we publish images to quay -- so the reason I'm trying to answer where should selinuxd run would also help me find a better home for the repo (should the project be part of the coreos org? openshift org? just be folded into SPO upstream?)

Thank you!

Reply:
selinuxd would probably also be useful for Fedora CoreOS too. Could you create an issue to discuss that there? Would love to see it in action in a demo.

Reply:
> selinuxd would probably also be useful for Fedora CoreOS too. Could you create an issue to discuss that there? Would love to see it in action in a demo.

I hope I found the right place - https://github.com/coreos/fedora-coreos-tracker/issues/854